apps/bfd-server-ng/src/main/java/gov/cms/bfd/server/ng/eob/EobHandler.java

```diff
@@ -61,8 +76,34 @@ public Bundle searchByBene(
     var eobs =
         claimRepository.findByBeneXrefSk(
             beneXrefSk.get(), serviceDate, lastUpdated, count, startIndex, sourceIds);
+
+    return buildEobBundleExcludingSamhsa(eobs, true);
+  }
+
+  /**
+   * Converts a list of {@link Claim} objects into a FHIR {@link Bundle} of Explanation of Benefits
+   * (EOBs). Optionally filters out claims that contain SAMHSA-coded procedures based on the
+   * provided flag.
+   *
+   * @param claims the list of {@link Claim} objects to be converted into FHIR EOBs
+   * @param excludeSamhsa a boolean flag indicating whether to filter out claims containing
+   *     SAMHSA-coded procedures
+   * @return a FHIR {@link Bundle} containing the converted Explanation of Benefits (EOBs)
+   */
+  protected Bundle buildEobBundleExcludingSamhsa(List<Claim> claims, boolean excludeSamhsa) {
+
+    var filteredClaims = claims;
+    if (excludeSamhsa) {
+      filteredClaims =
+          claims.stream()
+              // keep claims that do NOT contain any SAMHSA-coded procedures
+              .filter(claim -> !claimHasSamhsa(claim))
+              .toList();
```

Comment:
I'd prefer to avoid casting the final results if we can help it since it's easy to cause runtime exceptions with that approach. If we avoid calling `toList` and streaming the data twice, we can do that.

```suggestion
    var claimStream = claims.stream();
    if (excludeSamhsa) {
      claimStream =
              claimStream
              // keep claims that do NOT contain any SAMHSA-coded procedures
              .filter(claim -> !claimHasSamhsa(claim));
```